assets/js/components/wp-dashboard/WPDashboardPopularPages.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * WPDashboardPopularPages component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ANALYTICS, DATE_RANGE_OFFSET } from '../../modules/analytics/datastore/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import PreviewBlock from '../PreviewBlock';
+import ReportError from '../ReportError';
+import { numFmt } from '../../util';
+import TableOverflowContainer from '../../components/TableOverflowContainer';
+import { getDataTableFromData } from '../../components/data-table';
+import ReportZero from '../../components/ReportZero';
+import { isZeroReport } from '../../modules/analytics/util/is-zero-report';
+const { useSelect } = Data;
+
+const WPDashboardPopularPages = () => {
+	const dateRangeDates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		compare: true,
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+
+	const reportArgs = {
+		...dateRangeDates,
+		metrics: [
+			{
+				expression: 'ga:pageviews',
+				alias: 'Pageviews',
+			},
+		],
+		dimensions: [
+			'ga:pageTitle',
+			'ga:pagePath',
+		],
+		orderby: [
+			{
+				fieldName: 'ga:pageviews',
+				sortOrder: 'DESCENDING',
+			},
+		],
+		limit: 5,
+	};
+
+	const data = useSelect( ( select ) => select( MODULES_ANALYTICS ).getReport( reportArgs ) );
+	const error = useSelect( ( select ) => select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [ reportArgs ] ) );
+	const loading = useSelect( ( select ) => ! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [ reportArgs ] ) );
+
+	if ( loading ) {
+		return <PreviewBlock width="48%" height="92px" />;
```

Comment:
This preview component is incorrect here, which makes the loading state of the widget look odd.

Right here you're using the same like the other dashboard widget components, which however overall have different distribution and only take half the horizontal widget width. The component used here should match what is used in the legacy component `WPAnalyticsDashboardWidgetTopPagesTabl`, where it's a `<PreviewTable rows={ 6 } />`.

Reply:
Ah, good catch, sorry about that. I didn't see the loading component passed into the HOC right at the bottom of the file, and I must've used an example from elsewhere 🤦  

I've addressed this. I had to add a line of CSS to get the `PreviewTable` to display correctly because its parent was `display: flex`, but now it's looking the same as the legacy version from what I can see.